src/components/Prompt/Prompt.types.ts

```diff
@@ -0,0 +1,85 @@
+export interface PromptProps {
+  /**
+   * Determines if the prompt should be active. By default true.
+   * @default true
+   */
+  when?: boolean;
+  /**
+   * Creates the message to show for the blocked route, if any.
+   */
+  message: string | PromptMessageCallback;
+  /**
+   * Additional options for Modal Dialog. If options are not passed, browser's system dialog is used
+   */
+  modalOptions?: PromptDefaultModalProps;
+}
+
+export interface PromptBasicProps extends PromptProps {
+  /**
+   * `history` object of react-router
+   */
+  history?: PromptHistory;
+  /**
+   * Determines if the prompt should be active.
+   */
+  when: boolean;
+}
+
+export interface PromptModalProps extends Required<PromptProps> {
+  /**
+   * `history` object of react-router
+   */
+  history?: PromptHistory;
+}
+
+export interface PromptMessageCallback {
+  (location: any): boolean | string;
+}
+
+export interface PromptLocation {
+  pathname: string;
+  search: string;
+  hash: string;
+}
+
+export interface PromptHistory {
+  location: PromptLocation;
+  block: PromptHistoryBlockCallback;
+  push(path: string, state?: any): void;
+}
+
+export interface PromptHistoryBlockCallback {
+  (cb: (location: Location) => any): PromptCallback;
+}
+
+export interface PromptCallback {
+  (): boolean | string;
+}
+
+export interface PropDefaultModalOptions {
```

Comment:
1. do you need to export it?
2. the naming a bit confusing me. does `BasePromptModalOptions` look better for you?

Reply:
there is a type, it should be `PromptDefaultModalOptions`
does `PromptDefaultModalOptions` make more sense? these are options for default modal, provided by the Prompt component

Reply:
is it necessary to export?

Reply:
not at the moment, but it might be useful when using Prompt with modal, to pass modal options